### PR TITLE
Port to aglib: entry

### PR DESCRIPTION
**Pull request by patryk4815**
Changes:
- add decorator `@OnlyWithDbg` to make `gdb/lldb` only commands
- port `entry`

### CODE REVIEW

**Comment by disconnect3d** [comment]
@mbrla0 can I ask you to review this PR? :)

**Comment by mbrla0** [comment]
Alright. I'll do it in the afternoon.

File: pwndbg/commands/start.py [hunk 1]
```diff
@@ -9,25 +9,33 @@
...
 import pwndbg.commands
-import pwndbg.gdblib.elf
-import pwndbg.gdblib.symbol
+import pwndbg.dbg
 from pwndbg.commands import CommandCategory
-from pwndbg.dbg import EventType
+from pwndbg.dbg import BreakpointLocation
+
+if pwndbg.dbg.is_gdblib_available():
+    import gdb
```
Comment (mbrla0) [inline pwndbg/commands/start.py:22]: I don't really love that with `@OnlyWithDbg` we still need to do the whole song and dance of gating off imports depending on the debugger, but It works well enough, and I think it's a good enough solution, all things considered, so I guess it's fine?

**Comment by mbrla0** [comment]
Alrighty, looks okay to me, aside from my small gripe with import gating.

File: pwndbg/commands/start.py (hunk 1 above)
Comment (disconnect3d) [reply pwndbg/commands/start.py:22]: Why do we actually need import gdb there?

Comment (disconnect3d) [reply pwndbg/commands/start.py:22]: Perhaps we can do `if gdb := pwndbg.dbg.get_gdb()`?

Comment (patryk4815) [reply pwndbg/commands/start.py:22]: I tried do `get_gdb` but `mypy` complain

**Pull request merged**